### PR TITLE
Adds net-kourier

### PR DESCRIPTION
**Pull request by jmprusi**
**What this PR does, why we need it**:

Enables test and prow robot in knative/net-kourier

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

**Comment by chizhg** [comment]
/lgtm
/approve
/hold

/assign @chaodaiG 
Please unhold when you are ready to make the update.

**Comment by chaodaiG** [comment]
Please merge from latest master and run `make config` again

File: config/prow/testgrid/testgrid.yaml [hunk 1]
```diff
@@ -303,6 +303,28 @@ test_groups:
 - name: ci-knative-net-istio-test-coverage
   gcs_prefix: knative-prow/logs/ci-knative-net-istio-go-coverage
   short_text_metric: "coverage"
+- name: ci-knative-net-kourier-continuous
+  gcs_prefix: knative-prow/logs/ci-knative-net-kourier-continuous
+  alert_stale_results_hours: 3
+- name: ci-knative-net-kourier-nightly-release
+  gcs_prefix: knative-prow/logs/ci-knative-net-kourier-nightly-release
+  alert_options:
```
Comment (mattmoor) [inline config/prow/testgrid/testgrid.yaml]: Remove trailing whitespace:
```suggestion
  alert_options:
```

File: config/prow/testgrid/testgrid.yaml [hunk 2]
```diff
@@ -303,6 +303,28 @@ test_groups:
...
+  gcs_prefix: knative-prow/logs/ci-knative-net-kourier-continuous
+  alert_stale_results_hours: 3
+- name: ci-knative-net-kourier-nightly-release
+  gcs_prefix: knative-prow/logs/ci-knative-net-kourier-nightly-release
+  alert_options: 
+    alert_mail_to_addresses: "knative-productivity-dev@googlegroups.com"
+  num_failures_to_alert: 1
+- name: ci-knative-net-kourier-dot-release
+  gcs_prefix: knative-prow/logs/ci-knative-net-kourier-dot-release
+  alert_options:
```
Comment (mattmoor) [inline config/prow/testgrid/testgrid.yaml]: Remove trailing whitespace:
```suggestion
  alert_options:
```

File: config/prow/testgrid/testgrid.yaml [hunk 3]
```diff
@@ -303,6 +303,28 @@ test_groups:
...
+  num_failures_to_alert: 1
+- name: ci-knative-net-kourier-dot-release
+  gcs_prefix: knative-prow/logs/ci-knative-net-kourier-dot-release
+  alert_options: 
+    alert_mail_to_addresses: "knative-productivity-dev@googlegroups.com"
+  alert_stale_results_hours: 170
+  num_failures_to_alert: 1
+- name: ci-knative-net-kourier-auto-release
+  gcs_prefix: knative-prow/logs/ci-knative-net-kourier-auto-release
+  alert_options:
```
Comment (mattmoor) [inline config/prow/testgrid/testgrid.yaml]: Remove trailing whitespace:
```suggestion
  alert_options:
```

File: config/prow/testgrid/testgrid.yaml [hunk 4]
```diff
@@ -756,6 +778,35 @@ dashboards:
   - name: coverage
     test_group_name: ci-knative-net-istio-test-coverage
     base_options: "exclude-filter-by-regex=Overall$&group-by-directory=&expand-groups=&sort-by-name="
+- name: net-kourier
+  dashboard_tab:
+  - name: continuous
+    test_group_name: ci-knative-net-kourier-continuous
+    base_options: "sort-by-name="
+    alert_options:
```
Comment (mattmoor) [inline config/prow/testgrid/testgrid.yaml]: Remove trailing whitespace:
```suggestion
    alert_options:
```

File: config/prow/testgrid/testgrid.yaml [hunk 5]
```diff
@@ -756,6 +778,35 @@ dashboards:
...
+  - name: continuous
+    test_group_name: ci-knative-net-kourier-continuous
+    base_options: "sort-by-name="
+    alert_options: 
+      alert_mail_to_addresses: "knative-productivity-dev@googlegroups.com"
+    num_failures_to_alert: 3
+  - name: nightly
+    test_group_name: ci-knative-net-kourier-nightly-release
+    base_options: "sort-by-name="
+    alert_options:
```
Comment (mattmoor) [inline config/prow/testgrid/testgrid.yaml]: Remove trailing whitespace:
```suggestion
    alert_options:
```

File: config/prow/testgrid/testgrid.yaml [hunk 6]
```diff
@@ -756,6 +778,35 @@ dashboards:
...
+  - name: nightly
+    test_group_name: ci-knative-net-kourier-nightly-release
+    base_options: "sort-by-name="
+    alert_options: 
+      alert_mail_to_addresses: "knative-productivity-dev@googlegroups.com"
+    num_failures_to_alert: 1
+  - name: dot-release
+    test_group_name: ci-knative-net-kourier-dot-release
+    base_options: "sort-by-name="
+    alert_options:
```
Comment (mattmoor) [inline config/prow/testgrid/testgrid.yaml]: Remove trailing whitespace:
```suggestion
    alert_options:
```

File: config/prow/testgrid/testgrid.yaml [hunk 7]
```diff
@@ -756,6 +778,35 @@ dashboards:
...
+  - name: dot-release
+    test_group_name: ci-knative-net-kourier-dot-release
+    base_options: "sort-by-name="
+    alert_options: 
+      alert_mail_to_addresses: "knative-productivity-dev@googlegroups.com"
+    num_failures_to_alert: 1
+  - name: auto-release
+    test_group_name: ci-knative-net-kourier-auto-release
+    base_options: "sort-by-name="
+    alert_options:
```
Comment (mattmoor) [inline config/prow/testgrid/testgrid.yaml]: Remove trailing whitespace:
```suggestion
    alert_options:
```

File: config/prow/testgrid/testgrid.yaml (hunk 1 above)
Comment (jmprusi) [reply config/prow/testgrid/testgrid.yaml]: those changes were added by `make config` I guess, this should be changed on the template?

Comment (chaodaiG) [reply config/prow/testgrid/testgrid.yaml]: that's correct, feel free give it a shot

Comment (chaodaiG) [reply config/prow/testgrid/testgrid.yaml]: templates are here: https://github.com/knative/test-infra/tree/master/tools/config-generator/templates

Comment (chizhg) [reply config/prow/testgrid/testgrid.yaml]: This one cannot be fixed from the templates, it was due to the change in https://github.com/knative/test-infra/pull/1719 and I haven't figured out how to fix it.
Let's merge this PR first and I will take a further look.

**Comment by chizhg** [comment]
/lgtm
Please unhold when you are ready to make the update @chaodaiG

File: config/prow/testgrid/testgrid.yaml (hunk 1 above)
Comment (jmprusi) [reply config/prow/testgrid/testgrid.yaml]: I've been able to remove any trailing whitespace by modfiying the outputConfig function in the config generator toool. 

```go
// outputConfig outputs the given line, if not empty, to stdout.
func outputConfig(line string) {
	if strings.TrimSpace(line) != "" {
		fmt.Fprintln(output, strings.TrimRight(line, " "))
		emittedOutput = true
	}
}
```

not the nicest way of doing it perhaps... but, should I open a PR with those changes?

Comment (jmprusi) [reply config/prow/testgrid/testgrid.yaml]: well, just opened a PR with that change, feel free to close it if it's not the proper way of doing it: 
https://github.com/knative/test-infra/pull/1826

Comment (chaodaiG) [reply config/prow/testgrid/testgrid.yaml]: Let's wait for @chizhg to fix it, he is more familiar with it

**Pull request merged**